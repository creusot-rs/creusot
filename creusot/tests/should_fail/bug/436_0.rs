extern crate creusot_contracts;
use creusot_contracts::*;

struct S {
    g: Ghost<i32>,
}

#[ghost]
fn prophecy(x: &mut S) -> i32 {
    pearlite! { *(^x).g }
}

pub fn f() {
    let b = &mut S { g: gh! { 1i32 } };
    b.g = gh! { prophecy(b) + 1i32 };
    proof_assert! { false }
}
