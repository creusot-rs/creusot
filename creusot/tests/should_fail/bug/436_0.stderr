error[creusot]: called Logic function in Ghost context "creusot_contracts::__stubs::fin"
  --> 436_0.rs:10:5
   |
10 |     pearlite! { *(^x).g }
   |     ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `pearlite` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

